pkg/scalers/prometheus_scaler.go

```diff
@@ -37,6 +39,21 @@ type prometheusMetadata struct {
 	metricName    string
 	query         string
 	threshold     int
+
+	// bearer auth
+	enableBearerAuth bool
```

Comment:
Having these be boolean flags instead of a mode enum is a "hmm" thing. I could imagine using both client certs and one of the other two, but the original config options wouldn't allow for that. Might be better to use an enum here.

Reply:
@coderanger Thanks for the review!
I agree. I've implemented that this way to stay consistent with the metrics-api scaler (i.e. https://github.com/kedacore/keda/blob/main/pkg/scalers/metrics_api_scaler.go#L36) but I'm happy to change that into an enum here. Shall we adjust that in the metrics-api scaler as well?

Reply:
@marpio yeah, would be nice to adjust that in metrics-api scaler as well, but rather in a separate PR

Reply:
Are you gonna use the enum here?